std/file.d

```diff
@@ -2285,23 +2307,22 @@ version(Posix) @safe unittest
     Throws:
         $(D FileException) on error.
   +/
-version(StdDdoc) string readLink(C)(const(C)[] link) @safe;
-else version(Posix) string readLink(C)(const(C)[] link) @safe
+version(StdDdoc) string readLink(R)(R link)
+    if (isInputRange!R && isSomeChar!(ElementEncodingType!R) &&
+        !isConvertibleToString!R);
+else version(Posix) string readLink(R)(R link)
+    if (isInputRange!R && isSomeChar!(ElementEncodingType!R) &&
```

Comment:
Any reason `readLink` uses `ElementEncodingType` while `symlink` uses `ElementType`? It doesn't make a difference when passed to `isSomeChar`, but please settle on one.